### PR TITLE
Papercut 1114 jc nomenclature

### PR DESCRIPTION
**Pull request by geograham**
## Description

Addresses Papercut #1114 

## Checklist

I confirm that I have completed the following checks:

- [x] I have justified any large differences in the regression tests caused by this pull request in the comments.
- [x] I have added new tests where appropriate for the changes I have made.
- [x] If I have had to change any existing unit or integration tests, I have justified this change in the pull request comments.
- [x] If I have made documentation changes, I have checked they render correctly.
- [x] I have added documentation for my change, if appropriate.

### CODE REVIEW

File: process/io/obsolete_vars.py [hunk 1]
```diff
@@ -104,7 +104,10 @@
     "bcylth": None,
     "itfmod": None,
     "jcrit_model": None,
-    "jcritsc": None,
+    "jcritsc": "j_crit_sc",
+    "jcritstr": "j_crit_cable",
+    "jstrandoh_bop": "jcableoh_bop",
+    "jstrandoh_eof": "jcableoh_eof",
```
Comment (timothy-nunn) [inline process/io/obsolete_vars.py:110]: These variables do not need to be added here because they are not input variables.

**Review by timothy-nunn — CHANGES_REQUESTED** [review]
Happy with the code changes. I'll ask for comments from @ajpearcey @chris-ashe @stuartmuldrew about impact of renaming output variables. While its technically backwards incompatible, it might be _ok_ for these variables. 

I can see arguments on both sides, especially for the `strand` -> `cable` variables where the variable name does not match the quantity. Is that worth backwards incompatibility?

cc @jonmaddock

File: process/io/obsolete_vars.py (hunk 1 above)
Comment (geograham) [reply process/io/obsolete_vars.py]: Thanks, have removed.

**Review by timothy-nunn — APPROVED** [review]
Spoke with @chris-ashe who doesn't believe these changes will affect downstream scripts. Bluemira does not read these entries in our MFile.

While this is technically a breaking API change, I do not believe it will be in practice. Given the lack of comments and the fact these output names are misleading, I am happy to approve the change.

**Pull request merged**